Expanding DataPipe  to support DataFrames

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #71931

Differential Revision: [D37500516](https://our.internmc.facebook.com/intern/diff/D37500516)

VitalyFedyunin committed:
Expanding DataPipe  to support DataFrames

[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Expanding DataPipe  to support DataFrames"




[ghstack-poisoned]

ejguan commented on torch/utils/data/datapipes/iter/callable.py:
```diff
@@ -176,6 +208,18 @@ class CollatorIterDataPipe(MapperIterDataPipe):
-        collate_fn: Callable = _utils.collate.default_collate,
+        conversion: Optional[
+            Union[
+            Callable[[Any], Any],
```
```suggestion
            Callable[..., Any],
```

VitalyFedyunin replied:
Done

ejguan commented on torch/utils/data/datapipes/iter/callable.py:
```diff
@@ -131,6 +136,33 @@ def __setstate__(self, state):
 
 
+def _collate_helper(conversion, item):
+    # TODO(VitalyFedyunin): Verify that item is any sort of batch
```
Do we need to check the type of `item` here?

ejguan replied:
And, do we want to extend to support list of dictionary?

VitalyFedyunin replied:
As TODO line

ejguan commented on torch/utils/data/datapipes/iter/callable.py:
```diff
@@ -131,6 +136,33 @@ def __setstate__(self, state):
+def _collate_helper(conversion, item):
+    # TODO(VitalyFedyunin): Verify that item is any sort of batch
+    if len(item.items) > 1:
+        print(item.items)
```
nit: removing print

And, I am curious where does the attribute `items` come from? What is the actual type of `item`?

VitalyFedyunin replied:
`DataChunkDF`

ejguan reviewed (approved):
LGTM with a few questions.

Commit:
Update on "Expanding DataPipe  to support DataFrames"




[ghstack-poisoned]

Commit:
Update on "Expanding DataPipe  to support DataFrames"




[ghstack-poisoned]

Commit:
Update on "Expanding DataPipe  to support DataFrames"




[ghstack-poisoned]

Commit:
Update on "Expanding DataPipe  to support DataFrames"




[ghstack-poisoned]

Commit:
Update on "Expanding DataPipe  to support DataFrames"




[ghstack-poisoned]

Commit:
Update on "Expanding DataPipe  to support DataFrames"




[ghstack-poisoned]

Commit:
Update on "Expanding DataPipe  to support DataFrames"




[ghstack-poisoned]

VitalyFedyunin commented:
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37500516).

VitalyFedyunin committed:
Update on "Expanding DataPipe  to support DataFrames"


Differential Revision: [D37500516](https://our.internmc.facebook.com/intern/diff/D37500516)

[ghstack-poisoned]

VitalyFedyunin commented:
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37500516).

VitalyFedyunin commented:
@pytorchbot merge -g

VitalyFedyunin committed:
Update on "Expanding DataPipe  to support DataFrames"


Differential Revision: [D37500516](https://our.internmc.facebook.com/intern/diff/D37500516)

[ghstack-poisoned]

VitalyFedyunin commented:
Somehow internal tools lost approval. Need re-review.

VitalyFedyunin committed:
Update on "Expanding DataPipe  to support DataFrames"


Differential Revision: [D37500516](https://our.internmc.facebook.com/intern/diff/D37500516)

[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Expanding DataPipe  to support DataFrames"


Differential Revision: [D37500516](https://our.internmc.facebook.com/intern/diff/D37500516)

[ghstack-poisoned]

VitalyFedyunin commented:
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37500516).

VitalyFedyunin commented:
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37500516).

kit1980 commented on torch/utils/data/datapipes/dataframe/dataframes.py:
```diff
@@ -94,38 +124,127 @@ def __mul__(self, add_val):
+
+    def __init__(self, callable, ctx=None, **kwargs):
+        if ctx is None:
+            self.ctx = {'operations': [], 'variables': []}
```
@VitalyFedyunin This seems to violate inheritance properties as base class `Capture` has more data. Usually you should call the parent's init before own init, but here it wouldn't really work. Many classes in this file have this issue.
